src/main/java/mate/jdbc/dao/impl/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,115 @@
+package mate.jdbc.dao.impl;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.dao.ManufacturerDao;
+import mate.jdbc.dao.mappers.impl.ManufacturerMapper;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.DataSource;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    private static final Logger log = LogManager.getLogger(ManufacturerDaoImpl.class);
+    private ManufacturerMapper mapper = new ManufacturerMapper();
+
+    @Override
+    public Manufacturer create(Manufacturer element) {
+        String insertQuery
+                = "INSERT INTO manufacturers (name, country) VALUES (?, ?);";
+        try (Connection connection = DataSource.getConnection();
+                PreparedStatement statement = connection.prepareStatement(insertQuery,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, element.getName());
+            statement.setString(2, element.getCountry());
+            statement.executeUpdate();
+            try (ResultSet resultSet = statement.getGeneratedKeys()) {
+                resultSet.next();
+                element.setId(resultSet.getLong(1));
+                log.info("{} was created", element);
+                return element;
+            }
```

Comment:
```suggestion
            statement.executeUpdate();
            ResultSet resultSet = statement.getGeneratedKeys()
            if (resultSet.next()) {
                element.setId(resultSet.getLong(1));
                log.info("{} was created", element);
                return element;
            }
```

We don't need to close ResultSet explicitly as it will be closed after statement close (fix in all places)
https://docs.oracle.com/javase/7/docs/api/java/sql/ResultSet.html